src/testsets.jl

```diff
@@ -231,6 +231,7 @@ Recursively flatten `ts` to a vector of `TestSet`s.
 """
 function _flatten_results!(ts::AbstractTestSet)::Vector{<:AbstractTestSet}
     original_results = ts.results
+    has_new_properties = !isempty(something(properties(ts), Dict()))
```

Comment:
Not at all a significant performance concern but i believe this would work out better if
```suggestion
    has_new_properties = !isempty(something(properties(ts), tuple()))
```
as it will avoid allocating.